packages/compose/src/hooks/use-copy-on-click/index.js

```diff
@@ -30,45 +25,67 @@ export default function useCopyOnClick( ref, text, timeout = 4000 ) {
 		alternative: 'wp.compose.useCopyToClipboard',
 	} );
 
-	/** @type {import('react').MutableRefObject<Clipboard | undefined>} */
-	const clipboard = useRef();
 	const [ hasCopied, setHasCopied ] = useState( false );
 
 	useEffect( () => {
 		/** @type {number | undefined} */
 		let timeoutId;
+		/** @type Array<Element>) */
+		let triggers = [];
 
 		if ( ! ref.current ) {
 			return;
 		}
 
-		// Clipboard listens to click events.
-		clipboard.current = new Clipboard( ref.current, {
-			text: () => ( typeof text === 'function' ? text() : text ),
-		} );
+		// `triggers` is always an array, regardless of the value of the `ref` param.
+		if ( typeof ref.current === 'string' ) {
+			// expect `ref` to be a DOM selector
+			triggers = Array.from( document.querySelectorAll( ref.current ) );
+		} else if ( 'length' in ref.current ) {
+			// Expect `ref` to be a `NodeList`
+			triggers = Array.from( ref.current );
+		} else {
+			// Expect `ref` to be a single `Element`
+			triggers = [ ref.current ];
+		}
 
-		clipboard.current.on( 'success', ( { clearSelection, trigger } ) => {
-			// Clearing selection will move focus back to the triggering button,
-			// ensuring that it is not reset to the body, and further that it is
-			// kept within the rendered node.
-			clearSelection();
+		/**
+		 * @param {Event} e
+		 */
+		const copyTextToClipboard = ( e ) => {
+			const trigger = /** @type {HTMLElement | null} */ ( e.target );
+			const currentWindow = trigger?.ownerDocument.defaultView || window;
+			const textToCopy = typeof text === 'function' ? text() : text || '';
 
-			// Handle ClipboardJS focus bug, see https://github.com/zenorocha/clipboard.js/issues/680
-			if ( trigger ) {
-				/** @type {HTMLElement} */ ( trigger ).focus();
-			}
+			currentWindow?.navigator?.clipboard
+				?.writeText( textToCopy )
+				.then( () => {
+					// Clear selection and ensure focus stays on the trigger element.
+					if ( trigger ) {
+						currentWindow?.getSelection()?.removeAllRanges();
+
+						trigger.focus();
+					}
```

Comment:
The previous `clipboard.js` implementation used a hack to copy text to the clipboard which involved creating a temporary element, moving focus to it, selecting its text, and executing a "copy" command.

By switching to the new Clipboard APIs, clearing the selection and restoring focus may not be necessary — what do folks think?

Reply:
I agree. If that's the case, we shouldn't need the focus handling. It's probably worth double checking that focus stays on the button after clicking it still :)